client/state/partner-portal/reducer/partner.ts

```diff
@@ -41,31 +41,31 @@ export const activePartnerKey = withSchemaValidation(
 	},
 	( state = initialState.activePartnerKey, action: AnyAction ) => {
 		switch ( action.type ) {
-			case JETPACK_PARTNER_PORTAL_PARTNERS_ACTIVE_PARTNER_KEY_UPDATE:
+			case JETPACK_PARTNER_PORTAL_PARTNER_ACTIVE_PARTNER_KEY_UPDATE:
 				return action.partnerKeyId;
 		}
 
 		return state;
 	}
 );
 
-export const all = withoutPersistence( ( state = initialState.all, action: AnyAction ) => {
+export const current = withoutPersistence( ( state = initialState.current, action: AnyAction ) => {
 	switch ( action.type ) {
-		case JETPACK_PARTNER_PORTAL_PARTNERS_ALL_REQUEST_SUCCESS:
-			// Only store the partners with keys that are not disabled.
-			return filter( action.partners, ( partner ) => {
-				// Strip out disabled keys.
-				partner.keys = filter( partner.keys, ( key ) => key.disabled_on === null );
-				return partner.keys.length > 0;
-			} );
+		case JETPACK_PARTNER_PORTAL_PARTNER_REQUEST_SUCCESS:
```

Comment:
Similar to my above comment, should we move this logic into the API instead of the client?

Reply:
I think it's fine if the API still outputs a partner even if that partner does not have any valid keys as I think it's an important distinction to have between a normal WPCOM user and a WPCOM user who has a partner account albeit with no keys.